integration/deterministic-build-output-test.ts

```diff
@@ -0,0 +1,49 @@
+import { test, expect } from "@playwright/test";
+import globby from "globby";
+import fs from "fs";
+import path from "path";
+
+import { createFixtureProject, js } from "./helpers/create-fixture";
+
+test("builds deterministically under different paths", async () => {
+  let appInit = {
+    files: {
+      "app/routes/index.jsx": js`
+      import { json } from "@remix-run/node";
+      import { useLoaderData, Link } from "@remix-run/react";
+
+      export function loader() {
+        return json("pizza");
+      }
+
+      export default function Index() {
+        let data = useLoaderData();
+        return (
+          <div>
+            {data}
+          </div>
+        )
+      }
+    `,
+    },
+  };
+  let dir1 = await createFixtureProject(appInit);
+  let dir2 = await createFixtureProject(appInit);
```

Comment:
It looks like we don't actually need any custom files for this fixture. Would you mind changing `createFixtureProject` to allow `files` to be optional and then the entire `init` argument to be optional so then we can do this:

```suggestion
  let dir1 = await createFixtureProject();
  let dir2 = await createFixtureProject();
```

Reply:
> I'm surprised that test case fails without your changes

Indeed, the underlying bug is similarly surprising :) ... I verified it failed on `dev` 2 months ago, and again after rebasing yesterday

Reply:
> we don't actually need any custom files

I don't know if that's true; for the bug to manifest we need some routes and virtual modules. I can take a closer look though, perhaps I can simplify the files used

Reply:
Upon further testing, w/o any files you always still have root.tsx, so the cascading indeterminate hash problem still persists 👍

Reply:
elaborating slightly on why the test fails on `dev`, the virtual modules end up having different contents and thus different hashes, and so that cascades down resulting in different filenames/etc across what should be identical builds:
```
    Error: expect(received).toEqual(expected) // deep equality

    - Expected  - 2
    + Received  + 2

      Array [
        "build/index.js",
        "public/build/entry.client-JPJ6IQB3.js",
    -   "public/build/manifest-F783E782.js",
    -   "public/build/root-QVMFUD3S.js",
    +   "public/build/manifest-545DEB0B.js",
    +   "public/build/root-DZTWF5EQ.js",
        "public/build/_shared/chunk-RSZTKXZZ.js",
      ]
```

Reply:
Ok done 👍